### PR TITLE
`#unique_keys` returns the renamed column name

### PR DESCRIPTION
**Pull request by alpaca-tc**
### Motivation / Background

follow-up #46192

Fixed a bug where `unique_keys` returned the old column name after the column 
specified in add_unique_key was renamed.

### Detail

Since `pg_attribute.attname` may return the old column name after renaming a column,
match `attrelid, attnum` in the process of getting the list of column names.

### Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Changes that are unrelated should be opened in separate PRs.
* [x] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [x] Tests are added or updated if you fix a bug or add a feature.
* [ ] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.

### CODE REVIEW

**Comment by alpaca-tc** [comment]
I'll fix the missing test. 🙏

**Comment by yahonda** [comment]
> Since pg_attribute.attname may return the old column name after renaming a column,

Found similar discussions https://www.postgresql.org/message-id/5860814f-c91d-4ab0-b771-ded90d7b9c55%40www.fastmail.com

File: activerecord/lib/active_record/connection_adapters/postgresql/schema_statements.rb [hunk 1]
```diff
@@ -655,14 +655,10 @@ def unique_keys(table_name)
           SQL
 
           unique_info.map do |row|
-            deferrable = extract_constraint_deferrable(row["condeferrable"], row["condeferred"])
+            conkey = row["conkey"].tr("{}", "").split(",").map(&:to_i)
```
Comment (yahonda) [inline activerecord/lib/active_record/connection_adapters/postgresql/schema_statements.rb:658]:
```suggestion
            conkey = row["conkey"].delete("{}").split(",").map(&:to_i)
```

Comment (yahonda) [reply activerecord/lib/active_record/connection_adapters/postgresql/schema_statements.rb:658]: I prefer String#delete here for some minor readability.

Comment (alpaca-tc) [reply activerecord/lib/active_record/connection_adapters/postgresql/schema_statements.rb]: fixed and force pushed 👍

**Pull request merged**